### PR TITLE
Add memoize async util (rewrite get json lists in misc)

### PR DESCRIPTION
**Pull request by Bruception**
### Description

Usage

```ts
const asyncFunction = async () => {...};

const cachedAsyncFunction = memoizeAsync(asyncFunction);

await cachedAsyncFunction();
await cachedAsyncFunction(); // Cached
```

### CODE REVIEW

**Comment by Miodec** [comment]
Todo: 
 - [ ] Rewrite get json lists in misc to use this

File: frontend/src/ts/utils/misc.ts [hunk 1]
```diff
@@ -1244,6 +1267,30 @@ export async function sleep(ms: number): Promise<void> {
...
+  getKey?: (...args: Parameters<T>) => any
+): T {
+  const cache = new Map<any, Promise<ReturnType<T>>>();
+
+  return (async (...args: Parameters<T>): Promise<ReturnType<T>> => {
+    const key = getKey ? getKey.apply(args) : args[0];
+
+    if (cache.has(key)) {
+      const ret = await cache.get(key);
+      if (ret) {
```
Comment (Bruception) [inline frontend/src/ts/utils/misc.ts:1281]: IMO, this should check for explicit undefined

**Comment by Miodec** [comment]
@Bruception its ready, could you have a look through

**Comment by Bruception** [comment]
LGTM iiw

**Pull request merged**